src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -153,19 +168,20 @@ internal static unsafe partial class Utf8Utility
 
                         continue; // need to perform a bounds check because we might be running out of data
 
-                    Sse2LoopTerminatedEarlyDueToNonAsciiData:
+                    LoopTerminatedEarlyDueToNonAsciiData:
+                        // x86 can only be little endian, while ARM can be big or little endian
+                        // so if we reached this label we need to check both combinations are supported
+                        Debug.Assert((AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian) || Sse2.IsSupported);
 
-                        Debug.Assert(BitConverter.IsLittleEndian);
-                        Debug.Assert(Sse2.IsSupported);
 
                         // The 'mask' value will have a 0 bit for each ASCII byte we saw and a 1 bit
-                        // for each non-ASCII byte we saw. We can count the number of ASCII bytes,
+                        // for each non-ASCII byte we saw. trailingZeroCount will count the number of ASCII bytes,
                         // bump our input counter by that amount, and resume processing from the
                         // "the first byte is no longer ASCII" portion of the main loop.
+                        // We should not expect a total number of zeroes equal or larger than 16.
```

Comment:
>We should not expect a total number of zeroes equal or larger than 16. [](start = 27, length = 70)

minor: I would re-phrase the comment to "Make sure that pInputBuffer is not advanced by more than 15 positions."

Reply:
If that's ok with you, I would like to address this in another PR so that I don't reset the CI (it's taking a really long time to finish).
I have another similar TODO from the other PR (modifying a comment). We can collect similar requests and address them separately.

Reply:
Sure. No problem.